pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -232,6 +232,10 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			}
 		}
 
+		if apimgmtv3.ClusterConditionUpdated.IsFalse(cluster) && strings.HasPrefix(apimgmtv3.ClusterConditionUpdated.GetMessage(cluster), "[Syncing error") {
+			return cluster, nil
```

Comment:
I understand that coming back in may not do anything until the cluster has been updated, I can't help but feel it is appropriate to return the error here, for debugging purposes and because it is simply not in a good state. @cbron thoughts?

Reply:
There are two reasons why I didn't return the error here:
1. The error does not occur here. It is just being checked here.
2. If the Updated message begins with syncing error, then the syncer needs to run again or there must be action by the user to rectify the situation. That is, the user must either update the node groups or create new ones and delete old ones. This will not be fixed by returning an error here and re-entering the controller.

Reply:
I'm happy to change it if my thinking is incorrect. I was only answering your question.

Reply:
> If the Updated message begins with syncing error, then the syncer needs to run again or there must be action by the user to rectify the situation. That is, the user must either update the node groups or create new ones and delete old ones. This will not be fixed by returning an error here and re-entering the controller.

Would the user find out about it if you don't return the error here ?

Reply:
The error string is set as the message on the "Updated" condition and appears in the UI.

Reply:
I mainly like the persistence logs of returning the error even if it will not be able to fix itself

Reply:
It seems like we should return an error if the cluster is broken. The error was encountered in syncing but now this handler should error because it cannot work without the syncing working. At the very least syncing cron job will log the error but doing both might make sense.

Reply:
OK. The handler will return the error if it sees a syncing error.